http-protocol-binding.md

```diff
@@ -439,6 +445,119 @@ Content-Length: nnnn
 
 ```
 
+### 3.4. Multipart Content Mode
+
+In the _multipart_ content mode several events are sent into a single HTTP multipart
+request or response as described in [RFC2046, section 5.1][rfc2046-section-5.1]. 
+The _multipart_  content mode maps one event per part. Because each part consists in an 
+group of headers and a body, an event could be encoded following the same rules that applies
+to [Structured Content Mode](#32-structured-content-mode) and [Binary Content Mode](#31-binary-content-mode)
+within the part, respectively named _structured part_ and _binary part_.
+
+Every part of the HTTP envelope MUST be or a _structured part_ or a _binary part_.
+
+#### 3.4.1. HTTP Content-Type
+
+The [HTTP `Content-Type`][content-type] header MUST be set to the media type of `multipart/cloudevents` 
```

Comment:
I guess if we require this media type then my previous comment doesn’t make as much sense, however what if we just used multipart/mixed? Or do we really need to know we’re dealing with CEs in advance?

Reply:
How a receiver could distinguish between a multipart containing "generic stuff + cloudevents" and a multipart containing only "cloudevents". I think the same questions that i did in the comment above applies here.

Reply:
Why does a receiver need to know if there are CEs in there at all? Why can't it treat each part as an http message and if it happens to be a CE then it hands it to the CE processor.

Reply:
how sdks should handle a non CE part? should they discard it?

Reply:
what does it do today?

Reply:
An error, but i think the comparison with a single message doesn't hold here because the semantic is pretty clear: or i get an error or i get an event. What happens here?

Reply:
Isn't it just the same as with a single event? I would say that it depends on what the receiver expects. If you set up an endpoint that accepts only CloudEvents, you return an error. If your endpoint deals with mixed messages, it has to differentiate. 
Even if we define `multipart/cloudevents`, wouldn't an SDK have to be prepared for `multipart/mixed`, too? Do we exclude putting CloudEvents into a mixed multipart message?

Reply:
> Do we exclude putting CloudEvents into a mixed multipart message?

We could avoid excluding such use case, saying that you could eventually use this spec to send events in your own mixed multipart.

> wouldn't an SDK have to be prepared for multipart/mixed, too?

I don't think so. To avoid leaking/exposing the internals of the communication/http protocol, the SDKs needs somehow to distinguish between event and non-event. If we allow non cloudevents in a multi-cloudevent request/response, how such situation should be "modeled" by the sdk without exposing the internals? Does it just returns an error on the single part?